modules/swagger-jaxrs2/src/main/java/io/swagger/v3/jaxrs2/integration/OpenApiServlet.java

```diff
@@ -80,4 +106,43 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         }
 
     }
+
+    public static Map<String, List<String>> getQueryParams(String queryString) {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+
+        if (StringUtils.isEmpty(queryString)) {
+            return queryParameters;
+        }
+
+        String[] parameters = queryString.split("&");
+
+        for (String parameter : parameters) {
+            String[] keyValuePair = parameter.split("=");
+            queryParameters.put(keyValuePair[0], Collections.singletonList(keyValuePair[1]));
+        }
+        return queryParameters;
+    }
+
+    private static Map<String, String> getCookies(Cookie[] cookies) {
+        Map<String, String> mapOfCookies = new HashMap<>();
+        if (cookies != null) {
+            for (Cookie cookie : cookies) {
+                mapOfCookies.put(cookie.getName(), cookie.getValue());
+            }
+        }
+        return mapOfCookies;
+    }
+
+    public static Map<String, List<String>> getHeaders(HttpServletRequest req) {
+        if (req.getHeaderNames() == null) {
+            return new HashMap<>();
```

Comment:
Should we return `Collections.emptyMap()` here?

Reply:
I agree.